Puesdo Hooks/XComStrategyGame/XGFacility_Barracks/AddNewSoldier.txt

```diff
@@ -0,0 +1,34 @@
+//This script was generated by HexToPseudoCode decompiler for use with PatchUPK/PatcherGUI tool
+UPK_FILE = XComStrategyGame.upk
+OBJECT = XGFacility_Barracks.AddNewSoldier : AUTO
+[REPLACEMENT_CODE]
+
+/*
+                XComGameInfo(WorldInfo.Game).Mods[0].Mods("RoulettePlus.RoulettePlusMod").Object(kSoldier, true);
+                XComGameInfo(WorldInfo.Game).Mods[0].Mods("AllMods", "AddNewSoldier");
+*/
+
+			19 19 10 25 19 2E <XComGame.XComGameInfo> 19 01 <Engine.Actor.WorldInfo> [@] <Engine.WorldInfo.Game> 00 ( 01 <Engine.WorldInfo.Game> ) [@] <XComGame.XComGameInfo.Mods> 00 ( 01 <XComGame.XComGameInfo.Mods> ) [@] <NullRef> 00 ( 1B <Mods> 1F <%t "RoulettePlus.RoulettePlusMod"> 16 ) [@] <NullRef> 00 ( 1B <Object> 00 <.kSoldier> 27 16 )
+			19 10 25 19 2E <XComGame.XComGameInfo> 19 01 <Engine.Actor.WorldInfo> [@] <Engine.WorldInfo.Game> 00 ( 01 <Engine.WorldInfo.Game> ) [@] <XComGame.XComGameInfo.Mods> 00 ( 01 <XComGame.XComGameInfo.Mods> ) [@] <NullRef> 00 ( 1B <Mods> 1F <%t "AllMods"> 1F <%t "AddNewSoldier"> 16 )
+
+
+/*(0x0000/0x0000)*/ 49 [@] ( 28 15 ) 
+/*(0x0005/0x0005)*/ 49 [@] ( 28 15 ) 
```

Comment:
49 is the hex code for setting the default of optional parameters.
I'm not sure of the ramifications of this not being at the top of the function, code should very likely be under it.